### PR TITLE
Wiser, safer vendor includes

### LINKED ISSUE #76: Use new fastboot guard
> process.env.EMBER_CLI_FASTBOOT is soon going to be deprecated and [eventually removed](https://github.com/ember-fastboot/ember-cli-fastboot/issues/360). Therefore, we should instead use updateFastBootManifest hook.

https://github.com/ember-fastboot/ember-cli-fastboot#loading-additional-assets-in-fastboot-environment

### PR DESCRIPTION
**Pull request by jakeleboeuf**
This fixes #76- @kratiahuja, maybe you could give your 2 cents?

### CODE REVIEW

**Comment by kratiahuja** [comment]
We discussed offline over slack. Please move the css tree to `treeForStyles`. Also please make sure to test these scenarios too: https://gist.github.com/kratiahuja/d22de0fb1660cf0ef58f07a6bcbf1a1c#testing-guidelines

Rest LGTM.

File: index.js [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
+      path.join(this.project.root, 'node_modules', 'dropzone/dist/min'),
+      { files: ['dropzone.min.js'] }
+    );
+
+    dropzoneJs = map(
+      dropzoneJs,
+      content => `if (typeof FastBoot === 'undefined') { ${content} }`
+    );
+
+    return new MergeTrees(vendorTree, dropzoneJs);
```
Comment (kratiahuja) [inline index.js]: `new MergeTrees([vendorTree, dropzoneJs]);`

File: index.js [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
+      path.join(this.project.root, 'node_modules', 'dropzone/dist/min'),
+      {
+        files: ['dropzone.min.css'],
+        destDir: 'app/styles'
       }
-      return new MergeTrees(trees);
-    }
+    );
+
+    return new MergeTrees(styleTree, dropzoneCss);
```
Comment (kratiahuja) [inline index.js]: Same here.

Comment (jakeleboeuf) [reply index.js]: Seems trying to merge with a null tree throws an error `Expected Broccoli node, got undefined for inputNodes[0]`

**Comment by FutoRicky** [comment]
Hey, this looks good. Could you rebase with the latest changes on the dev branch? Thanks!

**Comment by FutoRicky** [comment]
Had to rebase with newer changes. Added these changes to most recent commit https://github.com/FutoRicky/ember-cli-dropzonejs/commit/d9355e31e6ca4ca4a4f0a78f64d14dcfaa1a0b5c

**Pull request closed without merging**